libs/agno/agno/agent/agent.py

```diff
@@ -1957,6 +1957,15 @@ def get_tools(
 
         # Add tools for accessing knowledge
         if self.knowledge is not None or self.retriever is not None:
+            # Check if retriever is an async function but used in sync mode
+            from inspect import iscoroutinefunction
+
+            if not async_mode and iscoroutinefunction(self.retriever):
```

Comment:
Lets rather use `from inspect import isawaitable`?

Reply:
- `isawaitable(obj)` checks if an object (the result of calling a function) can be awaited (i.e., is a coroutine, future, etc.).
- `iscoroutinefunction(fn)` checks if a function is an async function (i.e., defined with async def).
In the code, `self.retriever is a function`, not the result of calling the function.
- `isawaitable(self.retriever)` will always be False unless self.retriever is already a coroutine object (which it is not; it's a function).
- To use `isawaitable`, we would have to call the retriever (with dummy arguments), which i suppose would not be ideal.

Reply:
Thanks, you are right! I misunderstood how isawaitable is used.